src/tacacs_plus_fmt_plug.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Format for cracking TACACS+ hashes.
+ *
+ * https://insinuator.net/2015/06/tacacs-module-for-loki/
+ *
+ * This software is Copyright (c) 2015, Daniel Mende <dmende [at] ernw.de> and
+ * Copyright (c) 2017, Dhiru Kholia <dhiru [at] openwall.com>, and it is hereby
+ * released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
```

Comment:
This is weird. If we still have any code copyrighted by Daniel here, then let's keep only the second one of the two "Redistribution and ..." paragraphs - the whole thing would thus be under 3-clause BSD. However, I doubt there's any code by Daniel left here, and if so we can probably drop that copyright notice and use our usual 0-clause BSD with only Dhiru as the copyright holder. Indeed, we can and should nevertheless acknowledge Daniel for the inspiration or whatever it is that triggered or enabled our work.

Reply:
We still have the following borrowed code in our format,

``` C
if(((status >= 0x01 && status <= 0x07) || status == 0x21) && (flags == 0x01 || flags == 0x00) &&
        (6U + server_msg_len + data_len == cs->ctlen)) {
        return 1;
}
```

Good catch regarding the duplicate `Redistribution and use in source...` statement! I will fix it.

Reply:
Where exactly did this C code come from? I'm not sure it's subject to copyright. Credit for the idea, sure.

Reply:
The code comes from https://c0decafe.de/svn/codename_loki/trunk/.

Here is a blog post about it, https://insinuator.net/2015/06/tacacs-module-for-loki/.

> I'm not sure it's subject to copyright. 

I am not sure so I erred on the side of caution.

Reply:
What specific source file and lines did this come from? I only found relevant Python code there, not C.

Reply:
https://c0decafe.de/svn/codename_loki/trunk/lib/bf/tacacs.c

Reply:
Only the "status" check remains from the original code. I am not sure if this particular check can be expressed any differently (i.e. re-written).

Reply:
I took this to e-mail to Daniel, with Dhiru CC'ed.